CMakeLists.txt

```diff
@@ -230,7 +240,59 @@ cprover_default_properties(
     util
     xml)
 
+# Get the version of the java installation used by maven.
+#
+# The java in your PATH may not be the java used by maven.
+# Installing java and maven with `brew install openjdk maven` will install
+#   /opt/homebrew/opt/openjdk/bin/java
+#   /opt/homebrew/bin/mvn
+# and the cask-only installation directory /opt/homebrew/opt/openjdk/bin
+# will not be in your PATH by default.  See `brew info java`.
+#
+function(MAVEN_JAVA_VERSION VERSION)
+    # Output of mvn --version includes a string like "Java version: 16.1.4"
+    # Return the version number found in that string (like 16.1.4).
+    # Return UNKNOWN if the major version number can't be found.
+    execute_process(COMMAND mvn --version
+                    OUTPUT_VARIABLE STDOUT
+                    ERROR_VARIABLE STDERR
+                    RESULT_VARIABLE ERRNO
+                    OUTPUT_STRIP_TRAILING_WHITESPACE
+                    ERROR_STRIP_TRAILING_WHITESPACE)
+    string(REGEX REPLACE "[ \t\r\n]+" " " OUTPUT "${STDOUT}")
+    string(REGEX REPLACE "[ \t\r\n]+" " " ERROR "${STDERR}")
+    if(ERRNO)
+        message(STATUS "Warning: Failed to run mvn --version: ${ERROR}")
+        set(OUTPUT "Java version: ") # ensure VERSION=="UNKNOWN" below
+    endif()
+
+    string(REGEX REPLACE ".*Java version: ([0-9.]*).*" "\\1" FOUND ${OUTPUT})
+    if(FOUND STREQUAL OUTPUT)
+        message(STATUS
+                "Warning: Failed to find java version in mvn --version output: "
+                "${OUTPUT}")
+        set(FOUND "") # guarantee VERSION=="UNKNOWN" below
+    endif()
+
+    if(NOT FOUND)
+        set(FOUND "UNKNOWN")
+    endif()
+    set(${VERSION} ${FOUND} PARENT_SCOPE)
+endfunction()
+
+MAVEN_JAVA_VERSION(JAVA_VERSION)
+message(STATUS "Found Maven uses Java version ${JAVA_VERSION} by default")
+
 option(WITH_JBMC "Build the JBMC Java front-end" ON)
+if(NOT JAVA_VERSION VERSION_LESS 17)
```

Comment:
When https://github.com/Homebrew/homebrew-core/pull/85720/files gets merged to homebrew, `cbmc` (and `jbmc` by extension) will build cleanly with `openJDK` (version 17).

If you could drop the java version checks here (or at least, not deactivate `jbmc` building if `java --version` >= 17), then that would be great.